catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventToXmlElementConverter.java

```diff
@@ -194,4 +210,16 @@ public void addNamespace(String prefix, String uri) throws XMLStreamException {
         out.setPrefix(prefix, uri);
         namespaces.put(prefix, uri);
     }
+
+    private boolean checkNamespaceAdded(String uri) {
+
+        for (Iterator<Map<String, String>> reverseIter =
```

Comment:
Couldn't we use `addLast()` and `removeLast()` instead of `push()` and `pop()` (or use a simple `LinkedList` with `add()` and `removeLast()`)? That way we could use `scopeOfNamespacesAdded.stream().findAny()` here instead of a loop.